Updating previously captured transforms (#7023)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Internal changes ⚙️</h3>

* Quantum transform interpreters now perform argument validation and will no longer 

>   check if the equation in the `jaxpr` is a transform.

Reviewer: Maybe?
```suggestion
  check if the equation in the `jaxpr` is a transform primitive.
```

> * Interpreters for quantum transforms now validate arguments used by the transform and no longer remove them.
>   Furthermore, interpreters no longer check if the equation in the `jaxpr` is a transform.

Reviewer: Again, personal preference 😄 
```suggestion
* Quantum transform interpreters now perform argument validation and will no longer 
  check if the equation in the `jaxpr` is a transform.
```
